### PR TITLE
doc: Add page for signatures and crypto agility

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

add more docs.  i find the signature requirement is hard to find (the offline signing page is more of a practical and concrete example) so i created a new page for sui-signatures. it fits well to discuss crypto agility as well. 

## Test Plan 

n/a

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: doc/src/learn/cryptography/sui-offline-signing.md [hunk 1]
```diff
@@ -4,16 +4,15 @@ title: Offline Signing
 
 Sui supports offline signing, which is signing transactions using a device not connected to a Sui network, or in a wallet implemented in a different programming language without relying on the Sui key store. The steps to implement offline signing include:
 
- 1. Serialize the data for signing.
- 1. Sign the serialized data. Put the serialized data in a location to sign (such as the wallet of your choice, or tools in other programming languages) and to produce a signature with the corresponding public key.
- 1. Execute the signed transaction.
+1. Serialize the data for signing.
+2. Sign the serialized data. Put the serialized data in a location to sign (such as the wallet of your choice, or tools in other programming languages) and to produce a signature with the corresponding public key.
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-offline-signing.md:8]: Iirc there is a reason that we always put  - 1. (Rendering  automatically adds the counter) @randall-Mysten to confirm

File: doc/src/learn/cryptography/sui-signatures.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+# User Signature
+
+When a user submits a signed transaction, a serialized signature and a serialized transaction data is submitted. The serialized transaction data is the BCS serialized bytes of the struct `TransactionData` and the serialized signature is defined as a concatenation of bytes of `flag || sig || pk`.
+
+The `flag` is a 1-byte representation corresponding to the signature scheme that the signer chooses. The signing scheme and its corresponding flag are defined below:
+
+| Scheme | Flag |
+|---|---|
+|  Ed25519 |  0x00 |
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-signatures.md:17]: extra space?

File: doc/src/learn/cryptography/sui-signatures.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+1. The signature must be produced according to [RFC 8032](https://www.rfc-editor.org/rfc/rfc8032.html#section-5.1.6). The internal hash used is SHA-512.
+2. The signature must be valid according to [ZIP215](https://github.com/zcash/zips/blob/main/zip-0215.rst).
+
+See an concrete example for offline signing using CLI [here](../cryptography/sui-offline-signing.md).
+
+# Authority Signature
+
+The Authority on Sui (collection of validators) holds three distinctive keypairs:
+
+1. Protocol keypair
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-signatures.md:64]: Same, not sure if we have to set counters by ourselves cc @randall-Mysten

File: doc/src/learn/cryptography/sui-multisig.md [hunk 4]
```diff
@@ -1,37 +1,34 @@
...
-how to set up a local network, see [Sui Local Network](../build/sui-local-network.md).
+Pure Ed25519, ECDSA Secp256k1 and ECDSA Secp256r1 are supported as valid participating keys for MultiSig. A ([u8](https://doc.rust-lang.org/std/primitive.u8.html)) weight is set for each participating keys and the threshold can be set as [u16](https://doc.rust-lang.org/std/primitive.u16.html). If the serialized MultiSig contains enough valid signatures of which the sum of weights passes the threshold, the MultiSig is considered valid and the transaction can be executed. 
 
-## Applications of multi-signatures
+This topic discusses the applications of Multi-Signature and the workflow to create a Multi-Signature transaction in Sui.
 
-Multi-signature wallets offer several unique benefits to users, from increased security to escrow transactions and 2FA. 
-Unlike conventional multisig wallets, a Sui multisig account can accept weighted public keys from multiple key schemes.
-Currently, each key can be one of the following schemes: Ed25519, ECDSA secp256k1, and ECDSA secp256r1. 
+# Applications of Multi-Signature
```
Comment (joyqvq) [inline doc/src/learn/cryptography/sui-multisig.md:10]: @kchalkias should we talk about the application first then the cli example ? or we should flip the sections?

File: doc/src/learn/cryptography/sui-intent-signing.md [hunk 5]
```diff
@@ -46,7 +46,7 @@ The serialization of an `IntentMessage<T>` is the 3 bytes of the intent concaten
 
 ## User Signature
 
-To create a user signature, construct an intent message first, and create the signature over the BCS serialized value of the intent message (`intent || message`).
+To create a user signature, construct an intent message first, and create the signature over the Blake2b hash of the BCS serialized value of the intent message of the transaction data (`intent || message`).
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-intent-signing.md:49]: lets specifically mention the first 32 bytes. Realized many implementations out there output anything from 1..64 bytes.

File: doc/src/learn/cryptography/sui-multisig.md (hunk 4 above)
Comment (kchalkias) [reply doc/src/learn/cryptography/sui-multisig.md:10]: I was in a dilemma as well, for some reason I prefer the applications first, but let's have a good clear intro that we first present some applications and then documentation.

**Review by kchalkias — APPROVED** [review]

**Pull request merged**